src/main/java/at/hannibal2/skyhanni/features/itemabilities/abilitycooldown/ItemAbilityCooldown.kt

```diff
@@ -4,272 +4,259 @@ import at.hannibal2.skyhanni.SkyHanniMod
+//             message.contains("§lCASTING IN ") -> {
+//                 if (!ItemAbilityType.RAGNAROCK_AXE.isOnCooldown()) {
+//                     ItemAbilityType.RAGNAROCK_AXE.activate(LorenzColor.WHITE, 3.seconds)
+//                 }
+//             }
+//
+//             message.contains("§lCASTING") -> {
+//                 if (ItemAbilityType.RAGNAROCK_AXE.specialColor != LorenzColor.DARK_PURPLE) {
+//                     ItemAbilityType.RAGNAROCK_AXE.activate(LorenzColor.DARK_PURPLE, 10.seconds)
+//                 }
+//             }
+//
+//             message.contains("§c§lCANCELLED") -> {
+//                 ItemAbilityType.RAGNAROCK_AXE.activate(null, 17.seconds)
+//             }
         }
     }
 
-    private fun handleOldAbilities(message: String) {
-        // TODO use regex
-        if (message.contains(" (§6") && message.contains("§b) ")) {
-            val name: String = message.between(" (§6", "§b) ")
-            if (name == lastAbility) return
-            lastAbility = name
-            for (ability in ItemAbility.entries) {
-                if (ability.abilityName == name) {
-                    click(ability)
-                    return
-                }
-            }
-            return
-        }
-        lastAbility = ""
-    }
+//     private fun handleOldAbilities(message: String) {
+//         // TODO use regex
+//         if (message.contains(" (§6") && message.contains("§b) ")) {
+//             val name: String = message.between(" (§6", "§b) ")
+//             if (name == lastAbility) return
+//             lastAbility = name
+//             for (ability in ItemAbility.entries) {
+//                 if (ability.abilityName == name) {
+//                     click(ability)
+//                     return
+//                 }
+//             }
+//             return
+//         }
+//         lastAbility = ""
+//     }
 
     private fun isEnabled(): Boolean = LorenzUtils.inSkyBlock && config.itemAbilityCooldown
 
-    private fun click(ability: ItemAbility) {
-        if (ability.actionBarDetection) {
-            ability.activate()
-        }
-    }
-
     @SubscribeEvent
     fun onTick(event: LorenzTickEvent) {
         if (!isEnabled()) return
 
-        checkHotBar(event.isMod(10))
-    }
-
-    private fun checkHotBar(recheckInventorySlots: Boolean = false) {
-        if (recheckInventorySlots || abilityItems.isEmpty()) {
-            abilityItems = ItemUtils.getItemsInInventory(true).associateWith { hasAbility(it) }
+        for (activeAbility in activeAbilities.values) {
+            activeAbility.text = createItemText(activeAbility)
         }
-
-        items = abilityItems.mapValues { kp -> kp.value.map { createItemText(it) } }
     }
 
-    private fun createItemText(ability: ItemAbility): ItemText {
+    private fun createItemText(ability: ActiveAbility): ItemText {
         val specialColor = ability.specialColor
         val readyText = if (config.itemAbilityShowWhenReady) "R" else ""
```

Comment:
should be cached. Unnecessary branch. Make itemAbilityShowWhenReady a property and change the val on change.

Reply:
How to detect when changing without running through the logic? 
Or do you mean not running while ready until next item click? This is also not ideal as the mana check will be added in there as well later.

Reply:
Look at #957 
- https://github.com/hannibal002/SkyHanni/pull/957/files#diff-8e4b1a7a4e130932be9ee65f6c7d591ebd697c7f6afebffb9d3d6bea2e140749R18
- https://github.com/hannibal002/SkyHanni/pull/957/files#diff-2cac1551ced945996e6b5d2d5784d8aeb270be56323210b990334ca54799db62R162-R183

Reply:
probably over optimized, but since it is called a lot maybe this might be a good idea

Reply:
Oh you mean the readyText value?

Reply:
yes

Reply:
implemented